### PR TITLE
Cleanup parts referring to outdated stack

### PR DESCRIPTION
**Pull request by davidcassany**
This commit mostly removes references to outdated/dropped projects such
as rancherd, ros-installer and ros-operator. In addition it sliglty
adapts the surrounding text not to be completely wrong.

Note this is not a rewrite. It just tries to fix obsolete and even
wrong text, it is not aiming to provide missing contents.

Signed-off-by: David Cassany <dcassany@suse.com>

### CODE REVIEW

File: docs/architecture.md [hunk 1]
```diff
@@ -50,12 +49,12 @@ a generic Linux distribution.
...
 Elemental includes an operator that is responsible for managing OS upgrades
 and managing a secure device inventory to assist with zero touch provisioning.
-See the full operator docs at [Elemental-operator](https://github.com/rancher-sandbox/Elemental-operator/blob/main/README.md)
+See the project at [elemental-operator](https://github.com/rancher/elemental-operator/#readme)
 
-## Elemental Teal
+## The underlaying OS
 
-Elemental Teal is based off of SUSE Linux Enterprise (SLE) Micro for Rancher.  There is no specific dependency on
+Elemental is based off of SUSE Linux Enterprise (SLE) Micro for Rancher.  There is no specific dependency on
```
Comment (kkaempf) [inline docs/architecture.md:56]: This is not fully correct
- Elemental is the project/stack, (elemental, -toolkit, -operator, -cli, ...) which targets the Rancher stack - this is the "blue" part
- Elemental Teal is the combination of Elemental ("blue") with SLE Micro for Rancher ("green")

**Review by kkaempf — APPROVED** [review]
Let's merge as-is and iterate upon

File: docs/customizing.md [hunk 2]
```diff
@@ -40,31 +40,13 @@ check out your new image using docker with
...
-
-```bash
-bash ros-image-build myrepo/custom-build:v1.1.1 iso mycloud-config-file.txt
+bash elemental-iso-build CONFIG_FILE
 ```
 
-The third parameter is a path to a file that will be used as the cloud config passed to the installation.
-Refer to the [installation](./installation.md) and [configuration reference](./configuration.md) for the
-contents of the file.
+Where CONFIG_FILE is path to the configuration file including the registration data to register against the
```
Comment (fgiudici) [inline docs/customizing.md:51]:
```suggestion
Where CONFIG_FILE is the path to the configuration file including the registration data to register against the
```

**Review by fgiudici — APPROVED** [review]
LGTM!

File: docs/architecture.md [hunk 3]
```diff
@@ -1,10 +1,9 @@
 # Architecture
 
-Elemental is a toolkit to build an immutable Linux distribution.
+Elemental is an immutable Linux distribution.
```
[the remark this answers is not in the document]
Comment (fgiudici) [reply docs/architecture.md:3]: 🤔 From @kkaempf comment in the review:
> * Elemental is the project/stack, (elemental, -toolkit, -operator, -cli, ...) which targets the Rancher stack - this is the "blue" part
> * Elemental Teal is the combination of Elemental ("blue") with SLE Micro for Rancher ("green")

Seems to me Elemental Teal is our custom release of Elemental... well, honestly it looks like the same right now 🤔 😕 . Elemental *is* Elemental Teal right now, as the source code is based on SLE Micro and on our OBS images. 🤔

File: docs/architecture.md (hunk 1 above)
Comment (Itxaka) [reply docs/architecture.md:56]: what, why would you bring blue into the picture? Blue refers in toolkit to Fedora....way to make it more confusing :laughing:

File: docs/architecture.md [hunk 4]
```diff
@@ -30,7 +29,7 @@ image that is built using standard Docker build processes. Elemental is
 built using normal `docker build` and if you wish to customize the OS
 image all you need to do is create a new `Dockerfile`.
 
-## rancherd
+## Rancher system agent
```
Comment (Itxaka) [inline docs/architecture.md:32]: The below test is a bit incomplete. We dont only have the rancher-sytem-agent but we have several components integrated into EleTeal (Not gonna call  it elemental teal all the time LOL) that take care of the bootstrap. So I would mention the register which is the one that registers the node and the operator which takes care of the bootstrapping and letting system-agent take care of the rke/k3s stuff.

File: docs/configuration.md [hunk 5]
```diff
@@ -1,8 +1,12 @@
...
-that the config always starts with `#cloud-config`
+All custom configuration applied on top of a fresh deployment should come
+from a minimal `cloud-config` data. The `cloud-config` data can eventually
+be included within the OS image as a file in `/system/oem` or,
+alternatively, it can also be distributed from the Kubernetes management
+cluster as part of the machine registration data.
+
+Below is a reference of supported configuration. 
 
 ```yaml
```
Comment (Itxaka) [inline docs/configuration.md:11]: We should just have a decent example on the repo and call it from here to be included so its always updated

File: docs/upgrade.md [hunk 6]
```diff
@@ -20,7 +13,7 @@ TL;DR is
 kubectl edit -n fleet-local default-os-image
 ```
 ```yaml
-apiVersion: rancheros.cattle.io/v1
+apiVersion: elemental.cattle.io/v1
```
Comment (Itxaka) [inline docs/upgrade.md:16]: the osImage below is still pointing to the os2 version :o

File: docs/upgrade.md [hunk 7]
```diff
@@ -32,14 +25,13 @@ spec:
...
 
-For instance an upgrade channel file might look like this and is sufficient to `kubectl apply` it where the ros-operator is installed: 
+For instance an upgrade channel file might look like this and is sufficient to `kubectl apply` it to the Rancher management cluster:
+
 ```yaml
-apiVersion: rancheros.cattle.io/v1
+apiVersion: elemental.cattle.io/v1
 kind: ManagedOSVersionChannel
 metadata:
   name: os2-amd64
```
Comment (Itxaka) [inline docs/upgrade.md:37]: os2 name and below the REPOSITORY and IMAGE_PREFIX point to real os2 repos, I would change them for made up repos

**Review by Itxaka — COMMENTED** [review]
some comments

**Review by mjura — APPROVED** [review]
LGTM

File: docs/configuration.md (hunk 5 above)
Comment (davidcassany) [reply docs/configuration.md:11]: Agree, however this is not that straight forward. The example should come from YIP, as this is where this data is going to be de-serialized. I don't think we have a cloud-config (aka the supported cloud-init syntax) reference in YIP. I think we only have a reference for it's own syntax.

File: docs/upgrade.md (hunk 6 above)
Comment (davidcassany) [reply docs/upgrade.md:16]: well yes all this upgrade examples require a closer look and review, they could be easily really outdated. I'd suggest leaving this as it for now and fix it in #241

Another options is to remove most of it and leave it as a TODO or TBC.

File: docs/configuration.md (hunk 5 above)
Comment (Itxaka) [reply docs/configuration.md:11]: argh I thougth we had..that sucks, maybe we can send a patch to yip for that as it makes sense to have a reference.

File: docs/architecture.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (davidcassany) [reply docs/architecture.md]: As an example, I tried to use Rancher System Agent (no dashes and capitals) everywhere unless when referring the binaries in which cases I use `rancher-system-agent`.

**Review by Itxaka — APPROVED** [review]

**Pull request merged**